### PR TITLE
`Pagination` styling polish (HDS-1211 )

### PR DESCRIPTION
**Pull request by KristinLBradley**
### :pushpin: Summary
CSS polish and ~~responsive layout~~

**Preview:** https://hds-components-git-hds-1211-pagination-styling-polish-hashicorp.vercel.app/components/pagination#showcase

### :hammer_and_wrench: Detailed description
- Includes a few updates to documentation and accessibility tweaks
- This branch will be merged into the [Pagination component feature branch](https://github.com/hashicorp/design-system/pull/661) once approved. (It will not be merged directly into the main branch.)

### :camera_flash: Screenshots
**NOTE:** Truncation is not yet implemented. I will work on that concurrently to starting testing in Cloud-ui.

Screen Shot 2022-12-12 at 3 44 36 PM

### :link: External links
**Jira ticket:** [HDS-1211](https://hashicorp.atlassian.net/browse/HDS-1211)
**Figma file:** [visual design specs](https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?node-id=18709%3A42011&t=Y38GocLtWNy1s9iD-0)

***

### 👀 Reviewer's checklist:
Please review the CSS code and details of the visual UI.

~~I added experimental styles to stack the child components in small screen views and hide the Page Size Selector component. Let me know if I should adjust these styles or else just remove them. (According to @cveigt, the responsive layout is just an extra "nice to have" so not essential.)~~

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/app/styles/components/pagination.scss [hunk 1]
```diff
@@ -166,3 +166,25 @@ $pagination-height: calc(36px + 6px);
     background-position: center right 5px;
   }
 }
+
+// Responsive layout
+// max-width is based on where component layout breaks down currently, not finalized
```
Comment (KristinLBradley) [inline packages/components/app/styles/components/pagination.scss:171]: Note: If we want to keep these responsive layout styles then probably I would update the break point to use the “official” small screen size break point. The current one is just a loose experimental value. 

(I used a regular media query vs. a container query as the container query would require adding an extra wrapper div to apply styling to the container plus browser support is still pretty recent.)

**Review by MelSumner — APPROVED** [review]
Seems good! Might want to run axe (browser extension) on the documentation page, it's showing a few a11y issues (unrelated to this PR but related to the component).

File: packages/components/app/styles/components/pagination.scss (hunk 1 above)
Comment (didoo) [reply packages/components/app/styles/components/pagination.scss:171]: There are no "official" breaking points, unfortunately.

**Review by didoo — APPROVED** [review]
@KristinLBradley I suggest to merge this one, and I'll review the entire CSS in the main feature branch (#661)

File: packages/components/app/styles/components/pagination.scss (hunk 1 above)
Comment (KristinLBradley) [reply packages/components/app/styles/components/pagination.scss:171]: I understand, that's why I put quotes :) 

If there is something we are using in the website or something loose that's been discussed though I thought I could try to follow that if it made sense.

**Comment by KristinLBradley** [comment]
> Seems good! Might want to run axe (browser extension) on the documentation page, it's showing a few a11y issues (unrelated to this PR but related to the component).

**Comment by cveigt** [comment]
I notice that the buttons' height is higher than it is on the Figma component, which makes the active/selected indicator look significantly farther from the page number.

Screen Shot 2022-12-13 at 9 55 17 PM

Could we add cursor: not-allowed to the disabled state as we have for other interactive [components](https://hds-components-hashicorp.vercel.app/components/button#showcase)? This is helpful to colorblind users.

I have a question about functionality: Is there a way consumers can set different values to the SizeSelector? Since there are no prescriptive patterns (control over guidance) on items per page and the type of content paginated might vary, I think consumers might want to override the default options.

File: packages/components/app/styles/components/pagination.scss [hunk 2]
```diff
@@ -166,3 +166,25 @@ $pagination-height: calc(36px + 6px);
     background-position: center right 5px;
   }
 }
+
+// Responsive layout
+// max-width is based on where component layout breaks down
+@media screen and (max-width: 1000px) {
+  .hds-pagination {
+    display: flex !important;
```
Comment (didoo) [inline packages/components/app/styles/components/pagination.scss:174]: are you sure these `!important` are necessary?

File: packages/components/app/styles/components/pagination.scss [hunk 3]
```diff
@@ -166,3 +166,25 @@ $pagination-height: calc(36px + 6px);
     background-position: center right 5px;
   }
 }
+
+// Responsive layout
+// max-width is based on where component layout breaks down
+@media screen and (max-width: 1000px) {
```
[the remark this answers is not in the document]
Comment (didoo) [reply packages/components/app/styles/components/pagination.scss:172]: for example: if I am reading this file, I need to arrive to the end of the file to realize that there is a media query that changes the display of the `.hds-pagination` element, also though an `!important`, which means I have to restart to build my mental model of what the `.hds-pagination` is, how it works, etc.

File: packages/components/app/styles/components/pagination.scss [hunk 4]
```diff
@@ -166,3 +166,25 @@ $pagination-height: calc(36px + 6px);
...
+  .hds-pagination {
+    display: flex !important;
+    flex-direction: column;
+    height: auto !important;
+
+    > * {
```
Comment (didoo) [inline packages/components/app/styles/components/pagination.scss:178]: we should avoid using these kind of generic selectors and prefer to be more declarative/specific on what we're targeting.

Comment (KristinLBradley) [reply packages/components/app/styles/components/pagination.scss:178]: Ok but this is targeting flex children. I will change it.

**Comment by KristinLBradley** [comment]
> I notice that the buttons' height is higher than it is on the Figma component, which makes the active/selected indicator look significantly farther from the page number.
> 
> Screen Shot 2022-12-13 at 9 55 17 PM
> 
> Could we add `cursor: not-allowed` to the disabled state as we have for other interactive [components](https://hds-components-hashicorp.vercel.app/components/button#showcase)? This is helpful to colorblind users.
> 
> @KristinLBradley @didoo I have a question about functionality: Is there a way consumers can set different values to the SizeSelector? Since there are no prescriptive patterns (control over guidance) on items per page and the type of content paginated might vary, I think consumers might want to override the default options.

Thanks Christian, I'll take a look.

File: packages/components/app/styles/components/pagination.scss (hunk 4 above)
Comment (didoo) [reply packages/components/app/styles/components/pagination.scss:178]: I know but it's not an indeterminate or unknown list of children. we know which ones are the children we want to target

File: packages/components/app/styles/components/pagination.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (KristinLBradley) [reply packages/components/app/styles/components/pagination.scss]: Thanks for reviewing. I'm just going to remove any responsive layout related styling for now. We can re-examine later.

File: packages/components/app/styles/components/pagination.scss [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 @use "../mixins/focus-ring" as *;
 
-$pagination-height: calc(36px + 6px);
+$pagination-height: 36px;
```
Comment (KristinLBradley) [inline packages/components/app/styles/components/pagination.scss:8]: I'll add tokens for this and any other values needed after this.

File: packages/components/app/styles/components/pagination.scss [hunk 6]
```diff
@@ -131,8 +140,8 @@ $pagination-height: calc(36px + 6px);
     right: 0;
     bottom: 0;
     left: 0;
-    width: 19px;
-    height: 2px;
+    width: var(--token-pagination-indictor-width);
```
Comment (didoo) [inline packages/components/app/styles/components/pagination.scss:143]: typo: `indictor` → `indic[a]tor`

**Review by didoo — APPROVED** [review]
All good for me (only a typo to fix)

**Pull request merged**